lib/VM/Interpreter.cpp

```diff
@@ -2218,6 +2218,24 @@ CallResult<HermesValue> Interpreter::interpretFunction(
       }
 
       CASE(DeclareGlobalVar) {
+        // Check if an own-property already exists; if so do nothing.
+        // This may come about if the same BytecodeModule is run twice, for
+        // example.
+        NamedPropertyDescriptor ownDesc;
+        CAPTURE_IP_ASSIGN(
+            bool ownPropFound,
+            JSObject::getOwnNamedDescriptor(
+                runtime->getGlobal(),
+                runtime,
+                ID(ip->iDeclareGlobalVar.op1),
+                ownDesc));
+        if (ownPropFound) {
+          ip = NEXTINST(DeclareGlobalVar);
+          DISPATCH;
+        }
+
+        // We know that there is no such own-property; however the set may still
+        // fail (for example, if global is frozen).
         DefinePropertyFlags dpf =
             DefinePropertyFlags::getDefaultNewPropertyFlags();
```

Comment:
What if instead we don't set `dpf.setValue = 1`? Then the old value shouldn't be overwritten.

Reply:
Huh! I falsely believed that `setProperty=0` would *not* store a new property. A definite improvement!